### PR TITLE
Add 10 hours between shifts constraints to employee scheduling

### PR DESCRIPTION
**Pull request by Christopher-Chianelli**
### Referenced pull requests

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>full downstream build</b> please add the label `run_fdb`
* for a <b>specific pull request build</b> please add comment: <b>Jenkins (re)run [optaplanner-quickstarts] tests</b>
* for a <b>Quarkus LTS check</b> please add comment: <b>Jenkins run LTS</b>
* for a <b>specific Quarkus LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner-quickstarts] LTS</b>
* for a <b>Native check</b> please add comment: <b>Jenkins run native</b>
* for a <b>specific Native LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner-quickstarts] native</b>
</details>

### CODE REVIEW

File: use-cases/employee-scheduling/src/main/java/org/acme/employeescheduling/solver/EmployeeSchedulingConstraintProvider.java [hunk 1]
```diff
@@ -55,6 +57,17 @@ Constraint noOverlappingShifts(ConstraintFactory constraintFactory) {
                           EmployeeSchedulingConstraintProvider::getMinuteOverlap);
     }
 
+    Constraint atLeast10HoursBetweenTwoShifts(ConstraintFactory constraintFactory) {
+        return constraintFactory.forEachUniquePair(Shift.class,
+                                                   Joiners.equal(Shift::getEmployee),
+                                                   Joiners.lessThanOrEqual(Shift::getEnd, Shift::getStart))
+                .filter((firstShift, secondShift) -> firstShift.getEnd().until(secondShift.getStart(), ChronoUnit.HOURS) < 10)
```
Comment (ge0ffrey) [inline use-cases/employee-scheduling/src/main/java/org/acme/employeescheduling/solver/EmployeeSchedulingConstraintProvider.java]: Shall we use  ChronoUnit.HOURS.between(... end, ... start) instead?

File: use-cases/employee-scheduling/src/main/java/org/acme/employeescheduling/solver/EmployeeSchedulingConstraintProvider.java [hunk 2]
```diff
@@ -55,6 +57,17 @@ Constraint noOverlappingShifts(ConstraintFactory constraintFactory) {
...
+                .penalize("At least 10 hours between 2 shifts", HardSoftScore.ONE_HARD, (firstShift, secondShift) -> {
+                              int breakLength = (int) firstShift.getEnd().until(secondShift.getStart(), ChronoUnit.MINUTES);
```
Comment (ge0ffrey) [inline use-cases/employee-scheduling/src/main/java/org/acme/employeescheduling/solver/EmployeeSchedulingConstraintProvider.java]: Same here

**Review by ge0ffrey — APPROVED** [review]
Looks good. See that soft suggestion.

File: use-cases/employee-scheduling/src/main/java/org/acme/employeescheduling/solver/EmployeeSchedulingConstraintProvider.java (hunk 1 above)
Comment (Christopher-Chianelli) [reply use-cases/employee-scheduling/src/main/java/org/acme/employeescheduling/solver/EmployeeSchedulingConstraintProvider.java]: Decided to use `Duration.between(end, start).toHours()` instead

**Pull request merged**